### PR TITLE
isisimport, isisexport updates with new templates

### PR DESCRIPTION
**Pull request by scsides**
Add new import and export templates

## Description

Add new import templates for isisimport: Viking, LROC NAC, Cassini ISS, Cassini VIMS, Dawn FC, Dawn VIR, Kaguya MI, Kaguya TC, Lunar Orbiter HRC, MGS MOC, Mer MI, MRO CTX, Rosetta OSIRIS
Add new export templates for isisexport: LROC NAC, 

Reorganize isis/appdata/import directory with sub-directories for PDS3 and PDS4
Add comments to templates, isisexport and isisimport
Add cmake targets for new import directories
Use more inheritance in templates
Add template data JSON output to isisimport
Move JSON output in isisexport so render failures don't abort before writing the JSON
Many callbacks to isisimport
Target override to isisimport
Improve error messages
Add functional tests for isisimport

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [X] Documentation change (update to the documentation; no code change)
- [X] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] I have read and agree to abide by the [Code of Conduct](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/Code-Of-Conduct.md)
- [X] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [X] All new and existing tests passed.
- [X] I have added myself to the [.zenodo.json](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/.zenodo.json) document.
- [X] I have added any user impacting changes to the [CHANGELOG.md](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CHANGELOG.md) document.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [X] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

**Review by gsn9 — DISMISSED** [review]
Everything looks good to me, no comments.

File: isis/src/base/objs/PvlToJSON/PvlToJSON.cpp [hunk 1]
```diff
@@ -7,9 +7,11 @@ find files of those names at the top level of this repository. **/
 #include "PvlToJSON.h"
 
 #include <nlohmann/json.hpp>
+#include <QFile>
```
Comment (jessemapel) [inline isis/src/base/objs/PvlToJSON/PvlToJSON.cpp]: This looks like it's not needed. Was it required previously but included from another header before?

**Review by jessemapel — COMMENTED** [review]
Changes look good. There's no changelog entry. I can see an argument for not including this if it's not ready for user, but if we're merging it that means it's at least somewhat ready.

File: isis/src/base/objs/PvlToJSON/PvlToJSON.cpp (hunk 1 above)
Comment (scsides) [reply isis/src/base/objs/PvlToJSON/PvlToJSON.cpp]: Verified, not needed.

**Review by gsn9 — APPROVED** [review]

**Pull request merged**